packages/smooth_app/lib/data_models/user_preferences.dart

```diff
@@ -57,4 +60,11 @@ class UserPreferences extends ChangeNotifier {
 
   String get themeColorTag =>
       _sharedPreferences.getString(_TAG_THEME_COLOR_TAG) ?? 'COLOR_TAG_BLUE';
+
+  Future<void> setBottomTabIndex(final int index) async =>
+      _sharedPreferences.setInt(_TAG_BOTTOM_TAB_INDEX, index);
```

Comment:
I haven't read through the whole code yet, but I don't see the value in storing the index. Especially now where besides the scanner the other two pages don't provide that much value anymore.

Reply:
@M123-dev The point of storing the index is to use exactly the same navigation bar in all cases: the legacy cases (the former home page, divided in 3 tabs) and the new cases (purpose of this issue, and the product page is dealt with in this PR). And without too much complex code: just add `bottomNavigationBar: SmoothBottomNavigationBar()` to the `Scaffold`.

Reply:
Hey this PR is pretty close to my solution for this as well, so it looks good to me. 

I don't really see why we need the current index though? Is it so if a user closes the app and comes back, they land on the tab they we on when they closed the app ? In that case, it was discussed that whenever a user opens the app they should see the scan page with "Welcome to Open food facts" text on the search box.

Reply:
> it was discussed that whenever a user opens the app they should see the scan page with "Welcome to Open food facts" text on the search box

Oh, I didn't know that.
Behind saving the current index, there was also the question (already asked): which tab should look "selected" when we're not on one tab page in particular - e.g. when we are on the product page?

Reply:
So in a nutshell, this is what I understand the behavior to be:

- Navigation bar becomes sticky
- If the user goes into a tab and clicks around (example: opens the product page from history tab), he is still in that tab, just browsing
- Tabs are sticky, so lets say a user is on the "Product page" from the "History tab", he clicks on "Preferences tab", then clicks back on the "History tab", he will land on the "Product page" he was recently viewing, rather than on the "history page". Tapping again on the History tab will take the user to the "History page".  This behavior is akin to a few other apps like Gmail/parts of GMaps etc.
- When the user opens the app, the first tab displayed to the user is always the Scan page.

Reply:
> Navigation bar becomes sticky

More precisely, navigation bar was added to the product page. I'm not sure if there are other pages beyond profile, scan, history and product, but if so, the navigation bar should be added there too.

> If the user goes into a tab and clicks around (example: opens the product page from history tab), he is still in that tab, just browsing

No, because you asked me to remove the tab index from the preferences and that's where I kept its value. If you go to a product page, the navigation bar will display the default tab as selected: the scan tab. Anyway I didn't get the answer to "which tab should be selected when we are on the product page", and in the meanwhile I decided to select the scan tab.

> Tabs are sticky, so lets say a user is on the "Product page" from the "History tab", he clicks on "Preferences tab", then clicks back on the "History tab", he will land on the "Product page" he was recently viewing, rather than on the "history page". Tapping again on the History tab will take the user to the "History page". This behavior is akin to a few other apps like Gmail/parts of GMaps etc.

No, if you click on the history tab you always go to the history page.
This PR is only about issue #678, and does not implement the not-even-existing-yet "Make tabs stateful" issue evoked in #655.

> When the user opens the app, the first tab displayed to the user is always the Scan page.

That's correct, the default tab (which is opened in `main.dart`) is the scan tab.

Reply:
Hey, my previous message was more around generally what the navigation behavior should be like and not about how it would be after this PR. Sorry for misunderstanding. This PR adds the feature it's meant to add, so thank you for that.

> I'm not sure if there are other pages beyond profile, scan, history and product, but if so, the navigation bar should be added there too.

There's also the settings page, but that one is not that important, also doesn't feel like we should make the settings page sticky.

> "which tab should be selected when we are on the product page"
So if the user goes to a tab (lets say history tab) and clicks around on the page and lands on some other view (lets say product page), he is considered to still be on the history tab. I was thinking this should be doable without storing the tab index in user prefs. I'll take a stab at doing the rest.